gt/main.c

```diff
@@ -131,36 +129,78 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 		info->inner_ip_ver = ETHER_TYPE_IPv4;
 		info->l4_proto = inner_ipv4_hdr->next_proto_id;
 		info->l4_hdr = &inner_ipv4_hdr[1];
+
+		if (rte_ipv4_frag_pkt_is_fragmented(inner_ipv4_hdr)) {
+			info->frag = true;
+			info->l2_len = parsed_len;
+			info->l3_len = sizeof(*inner_ipv4_hdr);
+		}
 	} else if (likely(inner_ip_ver == 6)) {
-		/*
-	 	 * Make sure that the packet has space for
-		 * at least 4 bytes for the l4 header.
-	 	 */
-		if (pkt->data_len < parsed_len + sizeof(struct ipv6_hdr) + 4)
+		if (pkt->data_len < parsed_len + sizeof(struct ipv6_hdr))
 			return -1;
 
 		inner_ipv6_hdr = (struct ipv6_hdr *)info->inner_l3_hdr;
 		info->inner_ip_ver = ETHER_TYPE_IPv6;
 		info->l4_proto = inner_ipv6_hdr->proto;
 		info->l4_hdr = &inner_ipv6_hdr[1];
+
+		info->frag_hdr =
+			rte_ipv6_frag_get_ipv6_fragment_header(inner_ipv6_hdr);
```

Comment:
`rte_ipv6_frag_get_ipv6_fragment_header()` only checks the first extension header after the IPv6 header -- it will not check the whole extension header chain. Are we able to assume that the fragmentation header will be first?

Reply:
That's really a good point. If the assumption is not the case, I think we may have to patch the ``rte_ipv6_frag_get_ipv6_fragment_header()`` to locate the fragmentation header if any.

Reply:
Have you investigated this issue, @mengxiang0811? What did you find?